pkg/network/common/egressip.go

```diff
@@ -530,17 +531,22 @@ func (eit *EgressIPTracker) Ping(ip string, timeout time.Duration) bool {
 
 // Finds the best node to allocate the egress IP to, given the existing allocation. The
 // boolean return value indicates whether multiple nodes could host the IP.
-func (eit *EgressIPTracker) findEgressIPAllocation(ip net.IP, allocation map[string][]string) (string, bool) {
+func (eit *EgressIPTracker) findEgressIPAllocation(eip *egressIPInfo, allocation map[string][]string) (string, bool) {
 	bestNode := ""
 	otherNodes := false
 
 	for _, node := range eit.nodes {
 		if node.offline {
 			continue
 		}
+		if namespace, ok := eit.namespacesByVNID[eip.assignedVNID]; ok {
+			if sets.NewString(allocation[node.nodeName]...).HasAny(namespace.requestedIPs...) {
```

Comment:
This could maybe be a little less ad-hoc... like, you could keep track of the namespaces/VNIDs currently represented on a node in its `nodeEgress`?

or at least just split this out into a function with an obvious name like `nodeHasEgressIPForNamespace`

and anyway, it shouldn't be based on `namespace.requestedIPs` it should be based on `namespace.activeEgressIPs`. I think?